vm/src/stdlib/os.rs

```diff
@@ -793,6 +797,34 @@ mod _os {
             )
         }
 
+        #[cfg(not(unix))]
+        #[pymethod]
+        fn inode(&self, vm: &VirtualMachine) -> PyResult<u64> {
+            match self.ino.load() {
+                Some(ino) => Ok(ino),
+                None => {
+                    let stat = stat_inner(
+                        PathOrFd::Path(PyPathLike {
+                            path: self.entry.path(),
+                            mode: OutputMode::String,
+                        }),
+                        DirFd::default(),
+                        FollowSymlinks(false),
+                    )
+                    .map_err(|e| e.into_pyexception(vm))?
+                    .ok_or_else(|| vm.new_value_error("embedded null character".to_owned()))?;
+                    self.ino.swap(Some(stat.st_ino));
```

Comment:
Could you do a compare_exchange here? That way it won't get overwritten if a new thread calculated it in the meantime.